### DISCUSSION TITLE
Forced -T option for exec and run

### DISCUSSION THREAD

**Discussion by MarcusElevait** [PDMLab/docker-compose#274] [General]
Is there a reason, why the -T option for exec and run commands is always automatically added always? Shouldn't this be up to the user?

**Comment by AlexZeitler** [comment]
@MarcusElevait In the past, this option has been required to make this library work.

Local tests for the v2 version work for me without `-T` option.

#275 will validate it works in CI.

**Reply by AlexZeitler** [reply]
Seems to work on macOS (Intel based), too.

**Comment by MarcusElevait** [comment]
Thanks for the fast reply.
Yesterday i had the problem, that i could not do something like this:
`compose.exec('graphdb', 'bash -c "/init/scripts/reset-data.sh | console --serverURL http://localhost:7200"', dockerComposeOptions)`
Which runs perfectly fine, if i do it like that in the terminal
`docker compose exec graphdb bash -c '/init/scripts/reset-data.sh | console --serverURL http://localhost:7200'`

And my guess is, that it is because of the `-T` option.

**Reply by AlexZeitler** [reply]
I've just published a pre-release version of the #275 branch on npm: 

https://www.npmjs.com/package/docker-compose/v/0.24.9-0

Please let us know if this fixes your issue.

**Reply by MarcusElevait** [reply]
Wow, thanks for quickly providing this.
Unfortunately it does not solve our issue.
We still get this error:
`{"exitCode":2,"err":"|: -c: line 1: unexpected EOF while looking for matching `\"'\n","out":""}`

Any idea about this?

**Reply by AlexZeitler** [reply]
Please try something like this:

```shell
compose.exec('graphdb', ['bash', '-c', '/init/scripts/reset-data.sh', '|', 'console', '--serverURL', 'http://localhost:7200'],...)
```

**Reply by AlexZeitler** [reply]
Looks similar to this one: https://github.com/PDMLab/docker-compose/issues/116#issuecomment-648933269

**Reply by MarcusElevait** [reply]
If i do it like that, it just ignores the pipe and everthing after. It just executes the reset-data.sh file.

**Reply by AlexZeitler** [reply]
This?

```shell
compose.exec('graphdb', ['bash', '-c', '/init/scripts/reset-data.sh | console --serverURL http://localhost:7200'],...)
```

**Reply by MarcusElevait** [reply]
This works perfectly fine, thank you very much for your help, appreciate it :-)

**Reply by AlexZeitler** [reply]
:+1:
